subworkflows/local/input_check.nf

```diff
@@ -12,33 +12,20 @@ workflow INPUT_CHECK {
     SAMPLESHEET_CHECK ( samplesheet )
         .csv
         .splitCsv ( header:true, sep:',' )
-        .map { create_fastq_channel(it) }
+        .map { create_data_channel(it) }
```

Comment:
`create_fastq_channel` was used because it's got some `if` statements, etc. Your version is much simpler, so the meta map could be created on the fly:
```suggestion
        .map { [[id: it.sample, type: it.datatype], file(it.datafile), file(id.indexfile)] }
```